### PR TITLE
Setup cw-gitpod-base docker hub CI push

### PR DESCRIPTION
**Pull request by orkunkl**
`cw-gitpod-base` docker image for speeds up gitpod environment launch.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,52 +1,61 @@
...
+
+jobs:
+  docker-image:
+    executor: builder
     steps:
       - checkout
+      - setup_remote_docker
+          docker_layer_caching: true
       - run:
-          name: Version information
```
Comment (orkunkl) [inline .circleci/config.yml:10]: Deleted these, already excluded in `cargo-generate.toml`

File: .gitpod.yml [hunk 2]
```diff
@@ -1,5 +1,4 @@
-image:
-  file: .gitpod.Dockerfile
+image: cosmwasm/cw-gitpod-base:0.16
```
Comment (orkunkl) [inline .gitpod.yml:1]: Follows cosmwasm version

**Review by ethanfrey — APPROVED** [review]
lgtm

you are testing this manually on auto generated repo if I understand so feel free to merge when you have a nice run (or waiting for pebblenet?)

File: .circleci/config.yml (hunk 1 above)
Comment (webmaster128) [reply .circleci/config.yml:10]: `template.exclude` in `cargo-generate.toml` means the file does not go through the template engine but is shipped to the user directly.

**Comment by webmaster128** [comment]
This PR broke this promise

> We have template configurations for both GitHub Actions and Circle CI in the generated project, so you can get up and running with CI right away.

from https://github.com/CosmWasm/cosmwasm-template#ci-support

I wonder if that is really what we want.

**Pull request merged**